contracts/StableJoeStaking.sol

```diff
@@ -0,0 +1,327 @@
+
+        uint256 _previousAmount = user.amount;
+        uint256 _newAmount = user.amount.add(_amountMinusFee);
+        user.amount = _newAmount;
+
+        uint256 _len = rewardTokens.length;
+        uint256 _pending;
+        for (uint256 i; i < _len; i++) {
+            IERC20Upgradeable _token = rewardTokens[i];
+            updateReward(_token);
+
+            if (_previousAmount > 0) {
+                _pending = _previousAmount.mul(accRewardPerShare[_token]).div(PRECISION).sub(user.rewardDebt[_token]);
+                if (_pending != 0) safeTokenTransfer(_token, msg.sender, _pending);
+            }
+            user.rewardDebt[_token] = _newAmount.mul(accRewardPerShare[_token]).div(PRECISION);
+        }
+
+        internalJoeBalance = internalJoeBalance.add(_amountMinusFee);
+        joe.safeTransferFrom(msg.sender, feeCollector, _fee);
+        joe.safeTransferFrom(msg.sender, address(this), _amountMinusFee);
+        emit Deposit(msg.sender, _amountMinusFee, _fee);
+    }
+
+    /**
+     * @notice get user info
+     * @param _user The address of the user
+     * @param _rewardToken The address of the reward token
+     * @return uint256 the user's amount
+     * @return uint256 the user's reward debt of the reward token chosen
+     */
+    function getUserInfo(address _user, IERC20Upgradeable _rewardToken) external view returns (uint256, uint256) {
+        UserInfo storage user = userInfo[_user];
+        return (user.amount, user.rewardDebt[_rewardToken]);
+    }
+
+    /**
+     * @notice get the number of reward tokens
+     * @return The length of the array
+     */
+    function rewardTokensLength() external view returns (uint256) {
+        return rewardTokens.length;
+    }
+
+    /**
+     * @notice Set the reward token
+     * @param _rewardToken The address of the reward token
+     */
+    function addRewardToken(IERC20Upgradeable _rewardToken) external onlyOwner {
+        require(
+            !isRewardToken[_rewardToken] && address(_rewardToken) != address(0),
+            "StableJoeStaking: token can't be added"
+        );
+        require(rewardTokens.length < 25, "StableJoeStaking: list of token too big");
+        rewardTokens.push(_rewardToken);
+        isRewardToken[_rewardToken] = true;
+        updateReward(_rewardToken);
+        emit RewardTokenAdded(address(_rewardToken));
+    }
+
+    /**
+     * @notice Remove a reward token
+     * @param _rewardToken The address of the reward token
+     */
+    function removeRewardToken(IERC20Upgradeable _rewardToken) external onlyOwner {
+        require(isRewardToken[_rewardToken], "StableJoeStaking: token can't be removed");
+        updateReward(_rewardToken);
+        isRewardToken[_rewardToken] = false;
+        uint256 _len = rewardTokens.length;
+        for (uint256 i; i < _len; i++) {
+            if (rewardTokens[i] == _rewardToken) {
+                rewardTokens[i] = rewardTokens[_len - 1];
+                rewardTokens.pop();
```

Comment:
This works too, but I believe you can also do `delete rewardTokens[i]`?

Reply:
Correct, but then we need to reduce `rewardTokens.length` by 1, or the length of the array would still be the previous one.
Furthermore doing `rewardTokens.length--` is a really bad idea as it could overflow and return 2**256-1...
So we'd need to do something like this:
```
delete rewardTokens[i]
rewardTokens.length = rewardTokens.length.sub(1)
```

Reply:
Ok